### PR TITLE
address PG code review for StatefulGraphChatListClient

### PR DESCRIPTION
**Pull request by seekdavidlee**
// todo: handler - it's own class level method
// todo: test if we need this filter
// todo: we should check for duplicate ids and splice (remove dups)
// todo: asyc await on loadChatThreads

### CODE REVIEW

File: packages/mgt-chat/src/statefulClient/StatefulGraphChatListClient.ts [hunk 1]
```diff
@@ -203,40 +203,47 @@ class StatefulGraphChatListClient implements StatefulClient<GraphChatListClient>
...
+    latestChatThreads: ChatThreadCollection,
+    items: GraphChatThread[],
+    maxItems: number
+  ) {
+    const latestItems = latestChatThreads.value as GraphChatThread[];
+    const checkedItems = await this.checkWhetherToMarkAsRead(latestItems);
+
+    const idsIncheckedItems = new Set(checkedItems.map(item => item.id));
+    items = items.filter(item => !idsIncheckedItems.has(item.id));
+    items = items.concat(checkedItems);
```
Comment (plasne) [inline packages/mgt-chat/src/statefulClient/StatefulGraphChatListClient.ts:228]: This is going to re-order items. At a minimum we need to sort after doing this.

**Review by plasne — CHANGES_REQUESTED** [review]
Need to address sorting.

File: packages/mgt-chat/src/statefulClient/StatefulGraphChatListClient.ts (hunk 1 above)
Comment (seekdavidlee) [reply packages/mgt-chat/src/statefulClient/StatefulGraphChatListClient.ts:228]: done

**Comment by seekdavidlee** [comment]
> Need to address sorting.

done

**Review by plasne — APPROVED** [review]

**Review by TechnicallyWilliams — APPROVED** [review]

**Pull request merged**